core/modules/qproc/testQueryAnaOrderBy.cc

```diff
@@ -83,25 +91,69 @@ BOOST_AUTO_TEST_CASE(OrderByThreeField) {
+BOOST_AUTO_TEST_CASE(OrderByAggregate) {
+    std::string stmt = "SELECT objectId, AVG(taiMidPoint) "
+        "FROM Source "
+        "ORDER BY objectId ASC";
+    std::string expectedParallel = "SELECT objectId,COUNT(taiMidPoint) AS QS1_COUNT,SUM(taiMidPoint) AS QS2_SUM "
+                                   "FROM LSST.Source_100 AS QST_1_";
+    std::string expectedMerge = "SELECT objectId,(SUM(QS2_SUM)/SUM(QS1_COUNT))";
+    std::string expectedProxyOrderBy = "ORDER BY objectId ASC";
+    auto querySession = check(qsTest, stmt, expectedParallel, "", expectedMerge, expectedProxyOrderBy);
+}
+
+BOOST_AUTO_TEST_CASE(OrderByAggregateNotChunked) {
+    std::string stmt = "SELECT SUM(photClam) FROM Filter ORDER BY filterId";
+    std::string expectedParallel = "SELECT SUM(photClam) AS QS1_SUM FROM LSST.Filter AS QST_1_";
+    // FIXME merge query is not useful here, see DM-3166
+    std::string expectedMerge = "SELECT SUM(QS1_SUM)";
+    std::string expectedProxyOrderBy = "ORDER BY filterId";
+    auto querySession = check(qsTest, stmt, expectedParallel, "", expectedMerge, expectedProxyOrderBy);
+}
+
+BOOST_AUTO_TEST_CASE(OrderByLimit) {
+    std::string stmt = "SELECT objectId, taiMidPoint "
+        "FROM Source "
+        "ORDER BY objectId ASC LIMIT 5";
+    std::string expectedParallel = "SELECT objectId,taiMidPoint FROM LSST.Source_100 AS QST_1_ ORDER BY objectId ASC LIMIT 5";
+    std::string expectedMerge = "SELECT objectId,taiMidPoint ORDER BY objectId ASC LIMIT 5";
+    std::string expectedProxyOrderBy = "ORDER BY objectId ASC";
+    auto querySession = check(qsTest, stmt, expectedParallel, "", expectedMerge, expectedProxyOrderBy);
+}
+
+BOOST_AUTO_TEST_CASE(OrderByLimitNotChunked) { // Test flipped syntax in DM-661
     std::string bad = "SELECT run FROM LSST.Science_Ccd_Exposure limit 2 order by field";
     std::string good = "SELECT run FROM LSST.Science_Ccd_Exposure order by field limit 2";
-    std::string expected = "SELECT run FROM LSST.Science_Ccd_Exposure AS QST_1_ ORDER BY field LIMIT 2";
+    std::string expectedParallel = "SELECT run FROM LSST.Science_Ccd_Exposure AS QST_1_ ORDER BY field LIMIT 2";
+    std::string expectedProxyOrderBy = "ORDER BY field";
     // TODO: commented out test that is supposed to fail but it does not currently
-    // prepareTestQuerySession(qsTest, bad, "ParseException");
-    auto querySession = check(qsTest, good, expected);
-
-    auto mergeStmt = querySession->getMergeStmt();
-    if (mergeStmt) {
-        LOGF_INFO("mergeStmt->toString() %1%" % mergeStmt->toString());
-    }
-    else {
-        LOGF_INFO("No merge stmt");
-    }
+    // auto querySession = check(qsTest, bad, "ParseException");
+    auto querySession = check(qsTest, good, expectedParallel, "", "", expectedProxyOrderBy);
+}
+
+BOOST_AUTO_TEST_CASE(OrderByAggregateLimit) {
+    std::string stmt = "SELECT objectId, AVG(taiMidPoint) "
+        "FROM Source "
+        "ORDER BY objectId ASC LIMIT 2";
```

Comment:
It's valid MySQL syntax, and it test/cover the right part of the code, but I add "GROUP BY objectId" for better consistency.